FlowCal/mef.py

```diff
@@ -265,26 +319,30 @@ def fit_beads_autofluorescence(fl_channel, fl_mef):
     # Initial guesses:
     # 0: slope found by putting a line through the highest two points.
     # 1: y-intercept found by putting a line through highest two points.
-    # 2: bead autofluorescence initialized to 100.
+    # 2: bead autofluorescence initialized using the first point.
     params[0] = (np.log(fl_mef[-1]) - np.log(fl_mef[-2])) / \
-                    (fl_channel[-1] - fl_channel[-2])
-    params[1] = np.log(fl_mef[-1]) - params[0] * fl_channel[-1]
-    params[2] = 100.
+                (np.log(fl_au[-1]) - np.log(fl_au[-2]))
+    params[1] = np.log(fl_mef[-1]) - params[0] * np.log(fl_au[-1])
+    params[2] = np.exp(params[0]*np.log(fl_au[0]) + params[1]) - fl_mef[0]
 
     # Error function
     def err_fun(p, x, y):
-        return np.sum((np.log(y + p[2]) - ( p[0] * x + p[1] ))**2)
-        
+        return np.sum((np.log(y + p[2]) - ( p[0] * np.log(x) + p[1] ))**2)
+
     # Bead model function
     def fit_fun(p,x):
-        return np.exp(p[0] * x + p[1]) - p[2]
+        return np.exp(p[0] * np.log(x) + p[1]) - p[2]
 
-    # Channel-to-MEF standard curve transformation function
+    # a.u.-to-MEF standard curve transformation function
+    # We use the equivalent form ``exp(p[1]) * x**p[0]``. Mathematically, this
+    # is undefined for x < 0. In this case, we redefine the output as
+    # ``- exp(p[1]) * (-x)**p[0]``. These two can be combined as follows:
+    # ``np.sign(x) * np.exp(p[1]) * (np.abs(x)**p[0])``.
     def sc_fun(p,x):
-        return np.exp(p[0] * x + p[1])
+        return np.sign(x) * np.exp(p[1]) * (np.abs(x)**p[0])
```

Comment:
Yeah, reflection of the standard curve is pretty much what I did. 

Start from

```
s(x) = exp(p[1] + p[0]*log(x))
```

Which is defined for `x > 0` only. The following

```
s2(x) = exp(p[1]) * (x**p[0])
```

Is equivalent to the first form when `x > 0`, and it is also defined for `x = 0` as `s2(x) = 0`, for `p[0] > 0` (which I think should always be the case for us). Both `s(x)` and `s2(x)` converge to zero when `x -> 0+`, so `s2(x)` is continuous from the right.

I'm gonna guess that by "reflection", you meant making an odd function (https://en.wikipedia.org/wiki/Even_and_odd_functions#Odd_functions). I'll call this `s3(x)`. This function should satisfy `s3(x) = s2(x)` when `x >= 0`, and `s3(x) = -s2(-x)` when `x < 0`. In other words,

```
s3(x) =   exp(p[1]) * (  x **p[0]),    x >= 0
      = - exp(p[1]) * ((-x)**p[0]),    x < 0
```

From the discussion above, this is an odd function, continuous at zero with `s3(0) = s3(0+) = s3(0-) = 0`.

The definition of `s3(x)` can be combined in one single equation:

```
s3(x) = sign(x) * exp(p[1]) * (abs(x)**p[0])
```

Which is exactly what I have in the code.